### DISCUSSION TITLE
Why are there so many builds of Tesseract contained?

### DISCUSSION THREAD

**Discussion by rmast** [scribeocr/scribeocr#59] [Q&A]
When building, I see lot's of Tesseract versions being built, for example with and without legacy mode.
Isn't the version including legacy mode enough, just with some OEM parameter?

**Comment by Balearica** [comment]
Tesseract.js expects to see both lstm-only and legacy+lstm builds, which it picks between based on what settings are enabled.  By default, Tesseract.js only loads the LSTM code and data.  As the legacy model takes up more space, this behavior reduces total network traffic by 50-75% for most users (who use the default LSTM model).

The vast majority of Scribe OCR recognition modes use the legacy model, so having the LSTM-only build has fewer benefits within this codebase.  However, it is still included as Tesseract.js expects it to be there, and may try and load it if certain settings are selected.
